src/execution/physical_plan/plan_recursive_cte.cpp

```diff
@@ -18,13 +21,85 @@ unique_ptr<PhysicalOperator> PhysicalPlanGenerator::CreatePlan(LogicalRecursiveC
 	recursive_cte_tables[op.table_index] = working_table;
 
 	auto left = CreatePlan(*op.children[0]);
-	auto right = CreatePlan(*op.children[1]);
 
-	auto cte = make_uniq<PhysicalRecursiveCTE>(op.ctename, op.table_index, op.types, op.union_all, std::move(left),
-	                                           std::move(right), op.estimated_cardinality);
-	cte->working_table = working_table;
+	vector<LogicalType> payload_types, distinct_types;
+	vector<idx_t> payload_idx, distinct_idx;
+	vector<unique_ptr<BoundAggregateExpression>> payload_aggregates;
+
+	if (!op.key_targets.empty()) {
+		// create a group for each target, these are the columns that should be grouped
+		unordered_map<idx_t, idx_t> group_by_references;
+		for (idx_t i = 0; i < op.key_targets.size(); i++) {
+			auto &target = op.key_targets[i];
+			if (target->type == ExpressionType::BOUND_REF) {
```

Comment:
What happens if this is not true? Do we just ignore the key?

Reply:
That a key is not bounded should not occur. Perhaps an assertion would be the right choice here, or can you think of a scenario where the keys are not bound up to this point without throwing an error?